### PR TITLE
Example of creating a stereo agent

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

See #56 for context!

This one adds an example of creating an agent that uses a stereo RGB pair or a stereo depth pair.  The displaying to the user is done via cv2

## How Has This Been Tested

depth_example

rgb_example

Also via the new smoke test in pytest.
## Types of changes

-  New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: examples/stereo_agent.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene.id = (
+        "data/scene_datasets/habitat-test-scenes/skokloster-castle.glb"
+    )
+
+    # First, lets create a stereo RGB agent
+    left_rgb_sensor = habitat_sim.SensorSpec()
+    # Give it the uuid of left_sensor, this will also be how we
+    # index the observations to retreive the rendering from this sensor
```
Comment (msavva) [inline examples/stereo_agent.py]: typo: `retreive -> retrieve`

File: examples/stereo_agent.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    # Same deal with the right sensor
+    right_rgb_sensor = habitat_sim.SensorSpec()
+    right_rgb_sensor.uuid = "right_sensor"
+    right_rgb_sensor.resolution = [512, 512]
+    # The right RGB sensor will be 1.5 meters off the ground
+    # and 0.25 meters to the right of the center of the agent
+    right_rgb_sensor.position = 1.5 * habitat_sim.geo.UP + 0.25 * habitat_sim.geo.RIGHT
+
+    agent_config = habitat_sim.AgentConfiguration()
+    # Now we simly set the agents list of sensor specs to be the two specs for our two sensors
```
Comment (msavva) [inline examples/stereo_agent.py]: typo: `agents -> agent's`

File: examples/stereo_agent.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    agent_config.sensor_specifications = [left_rgb_sensor, right_rgb_sensor]
+
+    sim = habitat_sim.Simulator(habitat_sim.Configuration(backend_cfg, [agent_config]))
+
+    _render(sim, display)
+    sim.close()
+    del sim
+
+    # Now lets do the exact same thing but for a depth camera stereo pair!
```
Comment (msavva) [inline examples/stereo_agent.py]: typo: `lets -> let's`

File: examples/stereo_agent.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    left_depth_sensor = habitat_sim.SensorSpec()
+    left_depth_sensor.uuid = "left_sensor"
+    left_depth_sensor.resolution = [512, 512]
+    left_depth_sensor.position = 1.5 * habitat_sim.geo.UP + 0.25 * habitat_sim.geo.LEFT
+    # The only difference we need to do is set the sensor type to depth
```
Comment (msavva) [inline examples/stereo_agent.py]: grammar: `The only difference we need to do is set .. -> The only difference is we set ...`

File: examples/stereo_agent.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    left_depth_sensor.sensor_type = habitat_sim.SensorType.DEPTH
+
+    right_depth_sensor = habitat_sim.SensorSpec()
+    right_depth_sensor.uuid = "right_sensor"
+    right_depth_sensor.resolution = [512, 512]
+    right_depth_sensor.position = (
+        1.5 * habitat_sim.geo.UP + 0.25 * habitat_sim.geo.RIGHT
+    )
+    # The only difference we need to do is set the sensor type to depth
```
Comment (msavva) [inline examples/stereo_agent.py]: See above grammar comment

**Comment by erikwijmans** [comment]
> BGR vs RGB color channel flip

I am so bad at remembering to do that.....

File: examples/stereo_agent.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    for _ in range(100):
+        obs = sim.step("turn_right")
+        stereo_pair = np.concatenate([obs["left_sensor"], obs["right_sensor"]], axis=1)
+
+        if depth:
+            stereo_pair = np.clip(stereo_pair, 0, 10)
+            stereo_pair /= 10.0
+
+        if len(stereo_pair.shape) > 2:
+            stereo_pair = stereo_pair[..., 0:3][..., ::-1]
```
Comment (mathfac) [inline examples/stereo_agent.py]: As I understand `_render ` isn't considered part of example, but maybe still need more comments or const names for examples to make it clear for new people what's happening here:
`10.0` – threshold const and normalization function/variable name
`[..., 0:3][..., ::-1]` – need explanation

File: examples/stereo_agent.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+
+This can be done by giving the agent two sensors (be it RGB, depth, or semantic)
+with different positions.
+
+Note that the cameras must have different uuid's
+"""
+
+import random
+
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply examples/stereo_agent.py]: Yeah, that is how `isort` arranges things here :/

**Comment by msavva** [comment]
@erikwijmans since this PR has now been reviewed, shall we land it for release?

**Pull request merged**